code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/BasicTemplateNotificationBuilder.kt

```diff
@@ -0,0 +1,340 @@
+        )
+
+        // Create the notification channel if needed
+        channelIdToUse = createChannelAndGetChannelID(
+            context,
+            pushTemplate.channelId,
+            pushTemplate.sound,
+            pushTemplate.getNotificationImportance()
+        )
+
+        // create the notification builder with the common settings applied
+        val builder =
+            super.construct(context, pushTemplate, trackerActivityName, smallLayout, expandedLayout)
+
+        // add any action buttons defined for the notification
+        addActionButtons(
+            context,
+            trackerActivityName,
+            builder,
+            pushTemplate.actionButtonsString,
+            pushTemplate.tag,
+            pushTemplate.isNotificationSticky ?: false
+        )
+
+        // add a remind later button if we have a label and a timestamp
+        pushTemplate.remindLaterText?.let { remindLaterText ->
+            pushTemplate.remindLaterTimestamp?.let {
+                val remindIntent =
+                    createRemindPendingIntent(
+                        context,
+                        trackerActivityName,
+                        broadcastReceiverName,
+                        channelIdToUse ?: PushTemplateConstants.DEFAULT_CHANNEL_ID,
+                        pushTemplate
+                    )
+                builder.addAction(0, remindLaterText, remindIntent)
+            }
+        }
+
+        return builder
+    }
+
+    /**
+     * Adds action buttons for the notification.
+     *
+     * @param context the application [Context]
+     * @param trackerActivityName the [String] name of the activity to set in the created pending intent for tracking purposes
+     * @param builder the [NotificationCompat.Builder] to attach the action buttons
+     * @param actionButtonsString `String` a JSON string containing action buttons to attach
+     * to the notification
+     * notification
+     * @param tag `String` containing the tag to use when scheduling the notification
+     * @param stickyNotification [Boolean]  if false, remove the notification after the action
+     * button is pressed
+     */
+    fun addActionButtons(
+        context: Context,
+        trackerActivityName: String?,
+        builder: NotificationCompat.Builder,
+        actionButtonsString: String?,
+        tag: String?,
+        stickyNotification: Boolean
+    ) {
+        val actionButtons: List<AEPPushTemplate.ActionButton>? =
+            getActionButtonsFromString(actionButtonsString)
+        if (actionButtons.isNullOrEmpty()) {
+            return
+        }
+        for (eachButton in actionButtons) {
+            val pendingIntent: PendingIntent? =
+                if (eachButton.type === AEPPushTemplate.ActionType.DEEPLINK ||
+                    eachButton.type === AEPPushTemplate.ActionType.WEBURL
+                ) {
+                    BasicTemplateNotificationBuilder.createPendingIntent(
```

Comment:
Do we need to specify `BasicTemplateNotificationBuilder.createPendingIntent`? This method is not overridden in `BasicTemplateNotificationBuilder`, can we simply do `createPendingIntent`?

Reply:
Also can `createPendingIntent` accept `AEPPushTemplate.ActionButton`? If buttons for all notification types would need the same handling, we can move this logic of checking for action uri based on button type to a common place

Reply:
the `BasicTemplateNotificationBuilder` object was added mistakenly. the `createPendingIntent` function had moved classes a couple of times and this is a result of that.